src/google/protobuf/extension_set.h

```diff
@@ -269,6 +269,7 @@ class PROTOBUF_EXPORT ExtensionSet {
   void SetBool(int number, FieldType type, bool value, desc);
   void SetEnum(int number, FieldType type, int value, desc);
   void SetString(int number, FieldType type, const std::string& value, desc);
+  void SetString(int number, FieldType type, std::string&& value, desc);
```

Comment:
I believe it's possible to enable this optimization without creating new overloads. Could you try taking the existing methods and changing them to take a `std::string` (by value) instead of `const std::string&`? I think that should be backward-compatible with existing callers and still allow us to optimize the implementation.